### PR TITLE
JMPI-742-update-notification-component

### PR DESCRIPTION
**Pull request by issambaccouch**

### CODE REVIEW

File: JeMPI_Apps/JeMPI_Controller/src/main/java/org/jembi/jempi/controller/PsqlNotifications.java [hunk 1]
```diff
@@ -30,9 +31,9 @@ void insert(
 
          // Set auto-commit to false
          psqlClient.setAutoCommit(false);
-         Date res = new Date(created);
-         String state = "New";
-
+         Date date = new Date();
+         Timestamp res = new Timestamp(date.getTime());
+         String state = "OPEN";
          String sql = "INSERT INTO notification (id, type, state, names, created, patient_id, golden_id, score) "
```
Comment (walisc) [inline JeMPI_Apps/JeMPI_Controller/src/main/java/org/jembi/jempi/controller/PsqlNotifications.java:37]: Can we not use PrepareStatement for this? (helps with security/SQL injection prevention)

File: JeMPI_Apps/JeMPI_LibAPI/src/main/java/org/jembi/jempi/libapi/PsqlNotifications.java [hunk 2]
```diff
@@ -34,28 +33,38 @@ final class PsqlNotifications {
...
       final var list = new ArrayList<HashMap<String, Object>>();
       MatchesForReviewResult result = new MatchesForReviewResult();
       int skippedRows = 0;
       psqlClient.connect();
       try (PreparedStatement preparedStatement = psqlClient.prepareStatement(QUERY);
-           PreparedStatement countStatement = psqlClient.prepareStatement("SELECT COUNT(*) FROM notification")) {
+           PreparedStatement countStatement = psqlClient.prepareStatement("SELECT COUNT(*) FROM notification WHERE created BETWEEN ? AND ? AND state IN (?, ?, ?, ?)")) {
+         countStatement.setTimestamp(1, startDate);
+         countStatement.setTimestamp(2, endDate);
+         countStatement.setString(3, extractState(0, states));
```
Comment (walisc) [inline JeMPI_Apps/JeMPI_LibAPI/src/main/java/org/jembi/jempi/libapi/PsqlNotifications.java:53]: Can we not use `setArray` (https://docs.oracle.com/javase/8/docs/api/java/sql/PreparedStatement.html#setArray-int-java.sql.Array-)  instead. Also why 4 states, are we not only using `OPEN` and `CLOSE`

File: JeMPI_Apps/JeMPI_Controller/src/main/java/org/jembi/jempi/controller/PsqlNotifications.java (hunk 1 above)
Comment (issambaccouch) [reply JeMPI_Apps/JeMPI_Controller/src/main/java/org/jembi/jempi/controller/PsqlNotifications.java]: @walisc refactored. Thank you for the catch.

File: JeMPI_Apps/JeMPI_LibAPI/src/main/java/org/jembi/jempi/libapi/PsqlNotifications.java (hunk 2 above)
Comment (walisc) [reply JeMPI_Apps/JeMPI_LibAPI/src/main/java/org/jembi/jempi/libapi/PsqlNotifications.java:53]: I guess this is fine for now, but is there any reason why you didnt use `setArray` instead, and opted for the `extractState` approach?

File: JeMPI_Apps/JeMPI_LibMPI/src/main/java/org/jembi/jempi/libmpi/dgraph/DgraphQueries.java [hunk 3]
```diff
@@ -452,17 +463,13 @@ private static DgraphExpandedGoldenRecords searchGoldenRecords(
...
-
       String gql = "query search(" + String.join(", ", gqlArgs) + ") {\n";
       gql += String.format(Locale.ROOT, "all(%s) @filter(%s)", gqlFunc, gqlFilters);
       gql += "{\n";
       gql += CustomDgraphConstants.EXPANDED_GOLDEN_RECORD_FIELD_NAMES;
       gql += "}\n";
       gql += gqlPagination;
       gql += "}";
-
-      LOGGER.debug("Search Golden Records Query {}", gql);
```
Comment (walisc) [inline JeMPI_Apps/JeMPI_LibMPI/src/main/java/org/jembi/jempi/libmpi/dgraph/DgraphQueries.java:464]: Is there any reason why we are removing these debug logs?

File: JeMPI_Apps/JeMPI_UI/src/components/browseRecords/BrowseRecords.tsx [hunk 4]
```diff
@@ -31,13 +37,18 @@ import { useQuery } from '@tanstack/react-query'
...
+} from '@mui/x-date-pickers'
 import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs'
 import dayjs, { Dayjs } from 'dayjs'
 import getCellComponent from 'components/shared/getCellComponent'
 import { useNavigate, useSearchParams } from 'react-router-dom'
 import { Search } from '@mui/icons-material'
 import { useConfig } from 'hooks/useConfig'
+import CustomPagination from 'components/shared/CustomDataGridPagination'
 
 const getAlignment = (fieldName: string) =>
```
Comment (walisc) [inline JeMPI_Apps/JeMPI_UI/src/components/browseRecords/BrowseRecords.tsx:53]: What happens when they are field customizations. Does this cater for that?

File: JeMPI_Apps/JeMPI_UI/src/components/browseRecords/BrowseRecords.tsx [hunk 5]
```diff
@@ -51,15 +62,15 @@ const Records = () => {
...
-
-  const [dateFilter, setDateFilter] = useState(dayjs())
-
   const [dateSearch, setDateSearch] = useState(dayjs())
-
   const [searchParams, setSearchParams] = useSearchParams()
-
   const [isFetchingInteractions, setIsFetchingInteractions] = useState(
     searchParams.get('isFetchingInteractions')
       ? JSON.parse(searchParams.get('isFetchingInteractions') as string)
```
Comment (walisc) [inline JeMPI_Apps/JeMPI_UI/src/components/browseRecords/BrowseRecords.tsx:76]: Does this not result in an ambigous type i.e `string` or `boolean`. Why can't we convert this `isFetchingInteractions` to a boolean?

File: JeMPI_Apps/JeMPI_UI/src/components/browseRecords/BrowseRecords.tsx [hunk 6]
```diff
@@ -138,15 +141,14 @@ const Records = () => {
...
   useEffect(() => {
     setSearchParams(
       Object.entries(filterPayload).reduce(
         (acc, [k, v]) => {
           acc[k] = JSON.stringify(v)
           return acc
-          // eslint-disable-next-line @typescript-eslint/no-explicit-any
         },
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
         { isFetchingInteractions: isFetchingInteractions } as any
```
Comment (walisc) [inline JeMPI_Apps/JeMPI_UI/src/components/browseRecords/BrowseRecords.tsx:152]: very minor, but can we not just say `{ isFetchingInteractions }`

File: JeMPI_Apps/JeMPI_UI/package.json [hunk 7]
```diff
@@ -17,18 +17,25 @@
...
     "@emotion/styled": "^11.10.5",
     "@fontsource/roboto": "^4.5.8",
     "@mui/icons-material": "^5.10.9",
     "@mui/material": "^5.13.4",
+    "@mui/system": "^5.4.1",
     "@mui/x-data-grid": "^6.6.0",
     "@mui/x-date-pickers": "^6.6.0",
-    "@tanstack/react-query-devtools": "^4.16.1",
     "@tanstack/react-query": "^4.16.1",
+    "@tanstack/react-query-devtools": "^4.16.1",
```
Comment (walisc) [inline JeMPI_Apps/JeMPI_UI/package.json:34]: Should these, and the `@babel` items not be under `devDependencies`?

File: JeMPI_Apps/JeMPI_UI/src/components/dashboard/Dashboard.tsx [hunk 8]
```diff
@@ -83,80 +89,90 @@ const Dashboard = () => {
             label={<Typography variant="h5">M & U Values</Typography>}
             {...tabProps(1)}
           />
+          <Tab
+            label={<Typography variant="h5"> Import Process Status</Typography>}
+            {...tabProps(1)}
```
Comment (walisc) [inline JeMPI_Apps/JeMPI_UI/src/components/dashboard/Dashboard.tsx:94]: Should this not be `tabProps(2)`

File: JeMPI_Apps/JeMPI_UI/src/components/dashboard/widgets/ImportProcessWidget.tsx [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+      </Box>
+      <Box sx={{ minWidth: 35 }}>
+        <Typography variant="body2" color="text.secondary">{`${Math.round(
+          props.value
+        )}%`}</Typography>
+      </Box>
+    </Box>
+  )
+}
+export const ImportProcessWidget = () => {
```
Comment (walisc) [inline JeMPI_Apps/JeMPI_UI/src/components/dashboard/widgets/ImportProcessWidget.tsx:49]: Will need to update this later

File: devops/linux/docker/conf/postgres/config.sql [hunk 10]
```diff
@@ -20,7 +20,7 @@ CREATE TABLE IF NOT EXISTS Notification_State
 -- (
 --     Id uuid DEFAULT gen_random_uuid() PRIMARY KEY,
 --     Type_Id uuid,
---     Created date,
+--     Created timestamp without time zone,
```
Comment (walisc) [inline devops/linux/docker/conf/postgres/config.sql:23]: This needs to be moved to the bootstrapper app

**Review by walisc — APPROVED** [review]
Hi @ajinkyagadewar . Have reviewed this, and it looks generally fine. Made some minor updates if you want to have a look, else keen to get the merged as the M and U ui items depend on this branch.

**Comment by walisc** [comment]
Also, thanks for the work done on this @issambaccouch ! Like intergration of the ui dev mode to the docker swarm, and the record fetching optimizations! Quite cool! 🙏🏽

**Pull request merged**